pywbem_mock/_mainprovider.py

```diff
@@ -0,0 +1,4208 @@
+          * `host`: `None`, indicating the WBEM server is unspecified.
+
+
+        Raises:
+
+            CIMError: CIM_ERR_INVALID_NAMESPACE
+            CIMError: CIM_ERR_CLASS_NOT_FOUND
+        """  # noqa: E501
+        # pylint: enable:too-long
+
+        # Validate namespace
+        instance_store = self.get_instance_store(namespace)
+        class_store = self.get_class_store(namespace)
+
+        assert isinstance(ClassName, six.string_types)
+
+        # If DeepInheritance False use only properties from the original
+        # class. Modify property list to limit properties to those from this
+        # class.  This only works if class exists. If class not in
+        # CIM repository, ignore DeepInheritance.
+
+        # If None, set to server default
+        if DeepInheritance is None:
+            DeepInheritance = DEFAULT_DEEP_INHERITANCE
+
+        pl = PropertyList
+
+        # Get class property list which may be localonly or all
+        # superclasses
+        cl = self._get_class(namespace, ClassName, local_only=False)
+        class_pl = cl.properties.keys()
+
+        # If not DeepInheritance, compute property list to filter
+        # all instances to the properties in the target class as modified
+        # by the PropertyList
+        if not DeepInheritance:
+            if pl is None:  # properties in class form property list
+                pl = class_pl
+            else:      # reduce pl to properties in class_properties
+                pl_lower = [pc.lower() for pc in pl]
+                pl = [pc for pc in class_pl if pc.lower() in pl_lower]
+
+        # Create dictionary of all subclasses.
+        clns_dict = self._get_subclass_list_for_enums(ClassName, namespace,
+                                                      class_store)
+
+        # get and process instances from the instance_store
+        insts = [self._get_instance(namespace, inst.path, instance_store,
+                                    pl,
+                                    None,  # LocalOnly Not passed
+                                    IncludeClassOrigin,
+                                    IncludeQualifiers)
+                 for inst in instance_store.iter_values()
+                 if inst.path.classname in clns_dict]
+
+        return insts
+
+    ####################################################################
+    #
+    #   EnumerateInstanceNames server request
+    #
+    ####################################################################
+
+    def EnumerateInstanceNames(self, namespace, ClassName):
+        """
+        Implements a WBEM server responder for
+        :meth:`~pywbem.WBEMConnection.EnumerateInstanceNames`
+
+        Get instance names for instances that match the path defined
+        by `ClassName` and return a list of the names.
+
+        Parameters:
+
+          namespace (:term:`string`):
+            Name of the namespace in which the CIM instance will be created
+              (case independent).
```

Comment:
Copy & paste error from CreateInstance. Should use enumeration verbiage.

Reply:
I redid all of the text for the namespace parameter to be the same so this should be fixed.

**DONE**